### PR TITLE
Send foi request feature

### PR DESCRIPTION
**Pull request by EdwardAndress**
- User can input details on a basic form
- Some validations present
- Form is submitted and user sees confirmation
- No emails are sent (next feature!)

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
+
+resources :correspondence
```
Comment (aliuk2012) [inline config/routes.rb]: Could this be limited to just new and create actions?

```ruby
resources :correspondence, only: [:new, :create]
```

File: app/assets/stylesheets/correspondence.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Correspondence controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (aliuk2012) [inline app/assets/stylesheets/correspondence.scss]: Theres no need for this file, could just delete it

File: spec/controllers/correspondence_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe CorrespondenceController, type: :controller do
```
Comment (aliuk2012) [inline spec/controllers/correspondence_controller_spec.rb]: I know you've done a feature integration test but wondered if it was worth testings the actions in insolation. If you think we shouldn't then is there a need for this file?
http://everydayrails.com/2012/04/07/testing-series-rspec-controllers.html

**Comment by aliuk2012** [comment]
On Friday you mentioned concerns about naming:

>  I'm not sure about the naming...  correspondence#name sounds wrong, for example, but correspondence#sender_name lacks elegance.  correspondence#external_correspondent might work, then we have #internal_correspondent later on...  I'd suggest we leave it as it is for now though.

I'm happy to leave it as is for now as well. The one option I could think of was correspondence#requester_name, correspondence#requester_email etc. I came up with this naming convention by looking at the Trello board's "User" list and we defined those types of users as 

> Requesters: citizens, offenders, solicitors, staff, MPs, ministers, lords and journalists

File: spec/controllers/correspondence_controller_spec.rb (hunk 3 above)
Comment (EdwardAndress) [inline spec/controllers/correspondence_controller_spec.rb]: Good call.  I'll add some controller tests this morning.

**Comment by EdwardAndress** [comment]
@aliuk2012 - regarding naming:

I think 'requester' is OK but perhaps only works for FOI, where one makes requests, and not other forms of correspondence...

**Comment by EdwardAndress** [comment]
- reopened with changes from @aliuk2012 incorporated

**Pull request merged**